src/iron.rs

```diff
@@ -111,89 +128,139 @@ impl<H: Handler> Iron<H> {
     /// Kick off the server process using the HTTP protocol.
     ///
     /// Call this once to begin listening for requests on the server.
-    /// This consumes the Iron instance, but does the listening on
-    /// another task, so is not blocking.
-    ///
-    /// The thread returns a guard that will automatically join with the parent
-    /// once it is dropped, blocking until this happens.
-    pub fn http<A>(self, addr: A) -> HttpResult<Listening>
+    pub fn http<A>(self, addr: A) -> HttpResult<()>
         where A: ToSocketAddrs
     {
-        HttpListener::new(addr).and_then(|l| self.listen(l, Protocol::http()))
+        let addr = addr.to_socket_addrs()?.next().unwrap();
+
+        let core = Core::new().unwrap();
+        let handle = core.handle();
+
+        let sock = TcpListener::bind(&addr, &handle).unwrap().incoming();
+
+        return self.listen(sock, addr, Protocol::http(), core, handle);
     }
 
     /// Kick off the server process using the HTTPS protocol.
     ///
     /// Call this once to begin listening for requests on the server.
-    /// This consumes the Iron instance, but does the listening on
-    /// another task, so is not blocking.
-    ///
-    /// The thread returns a guard that will automatically join with the parent
-    /// once it is dropped, blocking until this happens.
-    pub fn https<A, S>(self, addr: A, ssl: S) -> HttpResult<Listening>
-        where A: ToSocketAddrs,
-              S: 'static + SslServer + Send + Clone
+    #[cfg(feature = "ssl")]
```

Comment:
@sfackler I'm really not sure how to handle this now.  It doesn't appear there is a generic way to get a ServerProto to start a server.

If we don't mind being forever bound to using TCP (something my current implementation isn't reliant on), then I could make a trait with a single method that takes a ```hyper::server::Http``` and returns something implementing ```ServerProto```, which I can then use with TcpServer.  Then https can provide a default version that uses tokio-tls and http can have a simple identity function.

Alternately, we could get tokio-proto to make a trait to start a generic server.  I'm not sure how useful/plausible that is.  We could make our own for now while that is investigated.

Thoughts?